src/server/forum_user_handler.cpp

```diff
@@ -184,95 +139,21 @@ fuh::ban_info fuh::user_is_banned(const std::string& name, const std::string& ad
 	//       prepared SQL statement API right now. However, they are basically
 	//       never used on forums.wesnoth.org, so this shouldn't be a problem
 	//       for the time being.
-	//
-
-	// NOTE: A ban end time of 0 is a permanent ban.
-	const std::string& is_extant_ban_sql =
-		"ban_exclude = 0 AND (ban_end = 0 OR ban_end >=" + std::to_string(std::time(nullptr)) + ")";
-
-	// TODO: retrieve full ban info in a single statement instead of issuing
-	//       separate queries to check for a ban's existence and its duration.
-
-	try {
-		if(!addr.empty() && prepared_statement<bool>("SELECT 1 FROM `" + db_banlist_table_ + "` WHERE UPPER(ban_ip) = UPPER(?) AND " + is_extant_ban_sql, addr)) {
+	ban_check b = conn.get_ban_info(name, addr);
+	switch(b.get_ban_type())
```

Comment:
I think there should be an explicit check if a ban was found. Using ban_type default of 0 as no ban is a bit too subtle. Or ban_type 0 should be specifically documented at least.

Reply:
I can add a code comment then.  Otherwise a separate query to check for a ban and then a near identical query to retrieve the information is kind of silly, and something like an additional `is_banned` boolean field doesn't tell anything that isn't already known.

Reply:
I'm not suggesting any change to the query. An additional `is_banned` flag or using a descriptive (enum) default value for one of the member variables that in principle cannot be returned from the query would make the code more readable.

As it stands someone can change the ban check query and add another type and it would count as the ban not being there at all. That's confusing and I would much rather get an assert or log message that an unexpected `ban_type` was returned.

That's why I would suggest to not use the `ban_type` for the validity check at all so you cannot accidentally choose the invalid value in your query and break the logic. If there is a value that `user_id` or so is not allowed to have then that value could be used but I think it's fine to add a dedicated flag for this.

Reply:
Alright, should be addressed now.